### DISCUSSION TITLE
files formats supported by the load command

### DISCUSSION THREAD

**Discussion by rodriguj-alexva** [skyfielders/python-skyfield#614] [Q&A]
Hello,
What file formats does the load() command support? I checked the manual but could not tell (I need to read STK ephemeris files).
Jose

**Comment by brandon-rhodes** [comment]
It supports `.bsp` files. There are also third-party libraries that support Type 1 and Type 21 ephemeris files. All those options can be seen in more detail here:

https://rhodesmill.org/skyfield/planets.html

The idea as described [here](https://ai-solutions.com/_help_Files/stk_ephemeris_file.htm) sounds pretty simple — just x,y,z coordinates in a file? I've never seen an STK file before, but it sounds like it wouldn't be hard to integrate into Skyfield, and if we were able to put code together then we could consider making it part of Skyfield itself once it was tested out.

**Reply by rodriguj-alexva** [reply]
The file format is just ASCII records with a header specifying epoch, type of records, plus other things. There are a number of different types of records. The files I have, use records w. 7 fields: relative time, pos_x, pos_y, pos_z, velocity_x, velocity_y, velocity_z separated by spaces. I already got a parser in Javascript and a Python one should not be hard. 
Can you point me to  say a parser in Skyfield to use a guide? (FYI I am only interested in importing ...)

**Reply by brandon-rhodes** [reply]
Most of the files that Skyfield reads are binary, so you might find other examples online that are more helpful that Skyfield's own code. Do you have an example file you can share?

**Reply by craigim** [reply]
Piping up because most of what I do with Skyfield is geared toward avoiding using STK.

The official spec for STK ephemeris is at the AGI website [here](https://help.agi.com/stk/index.htm#stk/importfiles-02.htm%3FTocPath%3DGetting%2520Started%7CCreate%2520%2526%2520Import%2520External%2520Files%7C_____8). They are basically time vs. position tables that STK can then interpolate to any time. If you add in things like velocity, you can get smoother interpolation. STK will both import and export data in these formats.
